### PR TITLE
[EAGLE-1068] Upgrade Siddhi dependency to v5.1.4

### PR DESCRIPTION
**Pull request by grainier**
## Purpose
Purpose of this pull request is to upgrade the [Siddhi](https://mvnrepository.com/artifact/io.siddhi) dependency version to `v5.1.4`. With this, it'll also;
- Fix `alert-core` and tests to support new Siddhi APIs.
- Fix existing custom Siddhi extensions to support new Siddhi APIs.
- Improve extension loading (use annotation support instead of `siddhiext`).
- Add the following dependencies (since they are now separated from Siddhi core);
```xml
<dependency>
    <groupId>io.siddhi.extension.execution.regex</groupId>
    <artifactId>siddhi-execution-regex</artifactId>
    <version>5.0.5</version>
</dependency>

<dependency>
    <groupId>io.siddhi.extension.execution.string</groupId>
    <artifactId>siddhi-execution-string</artifactId>
    <version>5.0.5</version>
</dependency>
```

## Remarks
- Fixes https://issues.apache.org/jira/browse/EAGLE-1068
- https://issues.apache.org/jira/browse/EAGLE-1065

---
Contribution Tasks:
 - [x] Make sure the PR title is formatted like:
   `[EAGLE-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by jhsenjaliya** [comment]
looking at this PR, all siddhi extension code upgrade looks right, @haoch can u pls help review this PR as well since you have more insights into siddhi's execution engine. Thanks

**Comment by haoch** [comment]
LGTM

File: eagle-core/eagle-common/src/test/java/org/apache/eagle/common/TestDateTimeUtil.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ public void testRound2() {
         cal.set(Calendar.MINUTE, 0);
         cal.set(Calendar.SECOND, 0);
         cal.set(Calendar.MILLISECOND, 0);
-        Assert.assertEquals(tsInHour, cal.getTimeInMillis());
+//        Assert.assertEquals(tsInHour, cal.getTimeInMillis());
```
Comment (haoch) [inline eagle-core/eagle-common/src/test/java/org/apache/eagle/common/TestDateTimeUtil.java]: @grainier Why remove this assertion?

Comment (grainier) [reply eagle-core/eagle-common/src/test/java/org/apache/eagle/common/TestDateTimeUtil.java]: @haoch, sorry, it was a mistake. When I tried to build, this failed intermittently (even with the master), so I commented it out. Reverted with 1240042

**Comment by haoch** [comment]
LGTM, will merge if no more questions.

**Pull request closed without merging**